src/string_buffer.rs

```diff
@@ -321,17 +351,45 @@ mod test {
         assert!(s.len == 4);
     }
 
-    // Test that truncating over a node panics.
-    // TODO this should actually work.
     #[test]
-    #[should_panic]
-    fn test_truncate_too_short() {
+    fn test_truncate_short() {
         let mut s: StringBuffer = StringBuffer::with_capacity(2);
         s.push_str("Ho");
         s.push_str(" world!");
         s.truncate(1);
-        //assert!(s.to_string() == "H");
-        //assert!(s.len == 1);
+        assert_eq!("H", s.to_string());
+        assert!(s.len == 1);
+    }
+
+    #[test]
+    fn test_truncate_noop() {
+        let mut s: StringBuffer = StringBuffer::with_capacity(2);
+        s.push_str("Ho");
+        s.truncate(5000);
+        assert_eq!("Ho", s.to_string());
+        assert!(s.len == 2);
+    }
+
+    #[test]
+    fn test_cur_offset_no_newlines() {
+        let mut s = StringBuffer::new();
+        s.push_str("Hello, World!");
+        assert_eq!(s.len, s.cur_offset());
+    }
+
+    #[test]
+    fn test_cur_offset() {
+        let mut s = StringBuffer::new();
+        s.push_str("Hello\nWorld! How goes it?");
+        assert_eq!(19, s.cur_offset());
+    }
+
+    #[test]
+    fn test_cur_offset_short() {
+        let mut s = StringBuffer::with_capacity(10);
+        s.push_str("Hello\nW");
+        s.push_str("orld! How goes it?");
+        assert_eq!(19, s.cur_offset());
     }
```

Comment:
You need tests for where there a multiple segments and the last newline is in segments other than the first (I don't think that is covered by these tests).

I'd also like to see tests which do multiple truncates and cur_offset after a truncate.